src/App/mermaidData/syncApiDataIntoOfflineStorage/SyncApiDataIntoOfflineStorage.js

```diff
@@ -6,6 +6,54 @@ const SyncApiDataIntoOfflineStorage = class {
 
   _dexiePerUserDataInstance
 
+  _getTableNamesWhereSyncingWasRejectedListedPerProject =
+    function _getTableNamesWhereSyncingWasRejectedListedPerProject(pushResponse) {
+      const projectsWithSyncErrors = {}
+
+      try {
+        const responseData = pushResponse.data
+
+        const mermaidApiSyncTableNames = Object.keys(responseData)
+
+        mermaidApiSyncTableNames.forEach((apiSyncTableName) => {
+          responseData[apiSyncTableName].forEach((itemSyncResponseObject) => {
+            const isSync403 = itemSyncResponseObject.status_code === 403
+            const syncErrorProjectName = itemSyncResponseObject.data.project_name
+            const syncErrorProjectId = itemSyncResponseObject.data.project_id
+
+            // to differentiate from other 403s, the proxy metric here is that if the
+            // response includes a project name, this means a user in not allowed to sync
+            const isUserDeniedFromSyncingToProject = isSync403 && syncErrorProjectName
```

Comment:
I would have liked the API to be more explicit that a user is being denied because they have been demoted, 403 + project info on the nested response object is good enough for now.

Reply:
ok good to know, regardless it helps to have clearer error messaging!